File: reference-app/src/main/java/org/smartregister/sample/fp/job/FPJobCreator.java
package org.smartregister.sample.fp.job;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import com.evernote.android.job.Job;
import com.evernote.android.job.JobCreator;

import org.smartregister.fp.common.job.ArchivedPostSterilizationJob;
import org.smartregister.job.ExtendedSyncServiceJob;
import org.smartregister.job.ImageUploadServiceJob;
import org.smartregister.job.PullUniqueIdsServiceJob;
import org.smartregister.job.SyncServiceJob;
import org.smartregister.job.SyncSettingsServiceJob;
import org.smartregister.job.ValidateSyncDataServiceJob;
import org.smartregister.sync.intent.SyncIntentService;

import timber.log.Timber;

public class FPJobCreator implements JobCreator {
    @Nullable
    @Override
    public Job create(@NonNull String tag) {
        switch (tag) {
            case SyncServiceJob.TAG:
                return new SyncServiceJob(SyncIntentService.class);
            case ExtendedSyncServiceJob.TAG:
                return new ExtendedSyncServiceJob();
            case ImageUploadServiceJob.TAG:
                return new ImageUploadServiceJob();
            case PullUniqueIdsServiceJob.TAG:
                return new PullUniqueIdsServiceJob();
            case ValidateSyncDataServiceJob.TAG:
                return new ValidateSyncDataServiceJob();
            case ViewConfigurationsServiceJob.TAG:
                return new ViewConfigurationsServiceJob();
            case SyncSettingsServiceJob.TAG:
                return new SyncSettingsServiceJob();
            case ArchivedPostSterilizationJob.TAG:
                return new ArchivedPostSterilizationJob();
            default:
                Timber.d("Looks like you tried to create a job " + tag + " that is not declared in the FP Job Creator");
                return null;
        }
    }
}

File: reference-app/src/main/java/org/smartregister/sample/fp/login/job/FPJobCreator.kt
package org.smartregister.sample.fp.login.job

import com.evernote.android.job.Job
import com.evernote.android.job.JobCreator
import org.smartregister.job.*
import org.smartregister.sync.intent.SyncIntentService
import timber.log.Timber

class FPJobCreator : JobCreator {
    override fun create(tag: String): Job? {
        return when (tag) {
            SyncServiceJob.TAG -> SyncServiceJob(SyncIntentService::class.java)
            ExtendedSyncServiceJob.TAG -> ExtendedSyncServiceJob()
            ImageUploadServiceJob.TAG -> ImageUploadServiceJob()
            PullUniqueIdsServiceJob.TAG -> PullUniqueIdsServiceJob()
            ValidateSyncDataServiceJob.TAG -> ValidateSyncDataServiceJob()
            ViewConfigurationsServiceJob.TAG -> ViewConfigurationsServiceJob()
            SyncSettingsServiceJob.TAG -> SyncSettingsServiceJob()
            else -> {
                Timber.d("Looks like you tried to create a job $tag that is not declared in the Anc Job Creator")
                null
            }
        }
    }
}